lib/scsi_defs.h

```diff
@@ -80,5 +87,60 @@ namespace scsi_defs {
     uint8_t vendorSpecific : 2;
   } ControlByte;
 
+  // SCSI Reference Manual Table 148 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  enum class SelectReport : uint8_t {
+    kRestrictedMethods = 0x0,
+    kWellKnown = 0x1,
+    kAllLogical = 0x2
+  };
+
+  // SCSI Reference Manual Table 147 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct ReportLunsCommand {
+    OptCode optCode = OptCode::kReportLuns;
+    uint8_t reserved1 : 8;
+    SelectReport selectReport : 8;
+    uint32_t reserved2 : 24;
+    uint32_t allocLength : 32;
+    uint8_t reserved3 : 8;
+    ControlByte controlByte;
+  } ABSL_ATTRIBUTE_PACKED; static_assert(sizeof(ReportLunsCommand) == 12);
+
+  // SCSI Reference Manual Table 149 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct ReportLunsData {
+    std::vector<LunAddress> lunList;
+
+    // Form arbitrarily sized vector of LUN lists.
+    ReportLunsData(uint8_t* byteData) {
+      uint8_t listLen = *((uint32_t*)byteData)/sizeof(LunAddress);
+      lunList.reserve(listLen);
+      for (uint32_t i = 0; i < listLen; ++i) {
+        uint32_t byteDataIndex = i * sizeof(LunAddress) + LUN_DATA_HEADER_SIZE;
+        LunAddress tmpLun = *(LunAddress*)(byteData + byteDataIndex);
+        lunList.push_back(tmpLun);
+      }
+    }
+
+    // Returns a vector of bytes describing the SCSI data.
+    std::vector<uint8_t> getValue() {
```

Comment:
Accessors [should be named like variables](https://google.github.io/styleguide/cppguide.html#Function_Names), i.e. snake_case. Also consider allowing the caller to pass in memory to write to, rather than allocating memory internally.